app/helpers/CYAChargeDHelper.scala

```diff
@@ -106,4 +109,249 @@ class CYAChargeDHelper(srn: String, startDate: LocalDate, accessType: AccessType
     )
   }
 
+  def isPsprForChargeD(index: Int, isPSR: Option[Boolean]): Seq[Row] = {
```

Comment:
let's move these common methods into another class and extend it by E and D and reuse the methods

Reply:
Created a common helper for common methods